lib/sdp/connection_data.ex

```
@@ -58,6 +58,56 @@ defmodule Membrane.Protocol.SDP.ConnectionData do
     end
   end
 
+  @spec serialize(sdp_address()) :: binary()
+  def serialize(address) when not is_list(address) do
+    with {:ok, serialized} <- serialize_address(address) do
+      serialized
+    end
+  end
+
+  @spec serialize([sdp_address()]) :: binary()
+  def serialize(addresses) do
+    case length(addresses) do
+      0 ->
+        ""
+
+      size ->
+        with {:ok, result} <- addresses |> Enum.sort_by(& &1.value) |> hd |> serialize_address do
+          "c=" <> result <> serialize_size(size)
+        end
+    end
+  end
+
+  defp serialize_size(1), do: ""
+  defp serialize_size(size) when size > 1, do: "/" <> Integer.to_string(size)
+
+  @spec serialize_address(%IP4{ttl: nil}) :: {:ok, binary()} | {:error, any()}
+  def serialize_address(%IP4{ttl: nil, value: value}) do
```

Are you sure you want to return `nil` from this function? You should always seek ways to fail fast so that `nil`s are not propagated (if they don't mean too much). I feel like in all clauses you might omit `with` and simply match.